tests/fixtures/language_container_fixture.py

```diff
@@ -0,0 +1,111 @@
+            pwd=itde.bucketfs.password)
+    container_file_path = Path(export_slc.cache_file)
+    with open(export_slc.cache_file, "rb") as fileobj:
+        container_bucketfs_location.upload_fileobj_to_bucketfs(
+            fileobj,
+            container_file_path.name)
+    with set_language_alias(flavor_path, itde, container_file_path) as language_alias:
+        wait_for_language_container_ready(itde, language_alias)
+    return container_file_path
+
+
+@pytest.fixture(scope="session")
+def language_alias(itde: TestConfig, flavor_path: Path, upload_slc: Path):
+    with set_language_alias(flavor_path, itde, upload_slc) as language_alias:
+        yield language_alias
+
+
+@contextlib.contextmanager
+def set_language_alias(flavor_path: Path, itde: TestConfig, container_file_path: Path):
+    release_name = container_file_path.with_suffix('').with_suffix('').name
+    language_definition = LanguageDefinition(
+        flavor_path=str(flavor_path),
+        bucketfs_name=bucketfs_params.name,
+        bucket_name=bucketfs_params.bucket,
+        add_missing_builtin=False,
+        path_in_bucket=bucketfs_params.path_in_bucket,
+        release_name=release_name
+    )
+    with revert_language_settings(itde.ctrl_connection):
+        language_alias = language_definition.generate_definition().split("=")[0]
+        itde.ctrl_connection.execute(language_definition.generate_alter_session())
+        itde.ctrl_connection.execute(language_definition.generate_alter_system())
+        yield language_alias
+
+
+def cleanup_images():
+    if "GITHUB_ACTIONS" in os.environ:
+        # Remove image and build output to reduce the disk usage in CI.
+        # We currently, use Github Actions as the CI and its disk is limited to 14 GB.
+        # TODO: This code can be removed if we moved to a CI with larger disks.
+        rm_docker_image = """docker images -a | grep 'transformers' | awk '{print $3}' | xargs docker rmi"""
+        subprocess.run(rm_docker_image, shell=True)
+
+
+def wait_for_language_container_ready(itde: TestConfig, language_alias: str):
+    schema = "upload_language_container"
+    udf_name = f"wait_{schema}"
+    itde.ctrl_connection.execute(f"CREATE SCHEMA If NOT EXISTS {schema}")
+    is_ready = False
+    wait_time_in_seconds = 180
+    for i in range(wait_time_in_seconds):
+        time.sleep(1)
+        is_ready = is_language_container_ready(itde, language_alias, schema, udf_name)
+        if is_ready:
+            break
+    if not is_ready:
+        raise Exception(f"Language container not ready after {wait_time_in_seconds}s.")
+
+
+def is_language_container_ready(itde: TestConfig, language_alias: str, schema: str, udf_name: str) -> bool:
+    try:
+        itde.ctrl_connection.execute(textwrap.dedent(f"""
+            CREATE OR REPLACE {language_alias} SCALAR SCRIPT {schema}.{udf_name}(i integer) 
+            RETURNS INTEGER AS
+                def run(ctx):
+                    return 1
+            / 
+            """))
```

Comment:
This creates the UDF every second. Wouldn't it be better to create the UDF once and then just call it every second?

Reply:
With creating it outside of the loop, there is the risk, that the container isn't yet present in the mounted bucketfs filesystem. Which trigger an error during the creation of the UDF.